source/tutorials/learning-journey/packaging-existing-software.md

```diff
@@ -0,0 +1,589 @@
+
+  src = builtins.fetchTarball {
+    url = "https://ftp.gnu.org/gnu/hello/hello-2.12.1.tar.gz";
+    sha256 = "0xw6cr5jgi1ir13q6apvrivwmmpr5j8vbymp0x6ll0kcv6366hnn";
+  };
+}
+```
+
+Now run the previous command again:
+
+```console
+$ nix-build -A hello
+this derivation will be built:
+  /nix/store/rbq37s3r76rr77c7d8x8px7z04kw2mk7-hello.drv
+building '/nix/store/rbq37s3r76rr77c7d8x8px7z04kw2mk7-hello.drv'...
+unpacking sources
+unpacking source archive /nix/store/xdbysilxxgbs55rrdxniglqg9m1v61h4-source
+source root is source
+patching sources
+configuring
+configure flags: --disable-dependency-tracking --prefix=/nix/store/y55w1djfnxkl2jk9w0liancp83zqb7ki-hello
+...
+configure: creating ./config.status
+config.status: creating Makefile
+...
+building
+build flags: SHELL=/nix/store/7q1b1bsmxi91zci6g8714rcljl620y7f-bash-5.2-p15/bin/bash
+... <many more lines omitted>
+```
+Great news: the derivation built successfully!
+
+The console output shows that `configure` was called, which produced a `Makefile` that was then used to build the project; it wasn't necessary to write any build instructions in this case, because the `stdenv` build system is based on `autoconf`, which automatically detected the structure of the project directory.
+
+### Build Result
+Check your working directory for the result:
+
+```console
+$ ls
+default.nix hello.nix  result
+```
+
+This result is a symbolic link to a Nix store location containing the built binary; you can call `./result/bin/hello` to execute this program:
+
+```console
+$ ./result/bin/hello
+Hello, world!
+```
+
+Congratulations, you have successfully packaged your first program with Nix!
+
+Next, you'll package another piece of software with external-to-`stdenv` dependencies that present new challenges, requiring you to make use of more `mkDerivation` features.
+
+## Something Bigger
+Now you will package a somewhat more complicated program, `icat`, which allows you to render images in your terminal.
+
+To start, modify the `default.nix` from the previous section by adding a new attribute for `icat`:
+
+```nix
+# default.nix
+let
+  pkgs = import <nixpkgs> { };
+in
+{
+  hello = pkgs.callPackage ./hello.nix { };
+  icat = pkgs.callPackage ./icat.nix { };
+}
+```
+
+Now copy `hello.nix` to a new file, `icat.nix`, and update the `name` attribute in that file:
+
+```nix
+# icat.nix
+{ lib
+, stdenv
+}:
+
+stdenv.mkDerivation {
+  name = "icat";
+
+  src = builtins.fetchTarball {
+	...
+  };
+}
+```
+
+Now to download the source code. `icat`'s upstream repository is hosted on [GitHub](https://github.com/atextor/icat), so you should slightly modify the previous [source fetcher](https://nixos.org/manual/nixpkgs/stable/#chap-pkgs-fetchers), this time using `pkgs.fetchFromGitHub` instead of `builtins.fetchTarball`, updating the argument attribute set to the function accordingly:
+
+```nix
+# icat.nix
+{ pkgs
+, lib
+, stdenv
+}:
+
+stdenv.mkDerivation {
+  name = "icat";
+
+  src = pkgs.fetchFromGitHub {
```

Comment:
```suggestion
{ lib
, stdenv
}:

stdenv.mkDerivation {
  name = "icat";

  src = fetchFromGitHub {
```

> > Don’t have pkgs in the arguments, bad practice.
>
> Why is this bad practice?

Because all `pkgs.<attr>` can be accessed by adding `<attr>` to the argument list. This then also allows people to override `<attr>` using `.override { <attr> = ...; }`